### PR TITLE
Fetch api

### PR DESCRIPTION
**Pull request by LisanneArdern**
In this PR I have fetched the API.

To get the app to function properly with API following things had to be adjusted:
- 2 fetches exported as hooks (one fetch for the search of crops and the other for the detailed view of each crop)
- use of isQuerying so that there is a loading screen while information is requested from API
- usage of useParams so that Pages can be reloaded 

**Please note that the test are currently not working so they can be ignored when reviewing my code. I will update them in my next PR.**

### CODE REVIEW

**Review by roscha80 — APPROVED** [review]
Looks good to me :) Good work!

File: src/pages/CropDetailsPage.js [hunk 1]
```diff
@@ -52,20 +58,41 @@ export default function CropDetailsPage({
...
           <h2>Details</h2>
-          <p>{details}</p>
+          <p>
+            {details !== null ? (
+              details
+            ) : (
+              <span>
+                Unfortunately, the description of this crop is not yet
+                available. Please be patient and check back later, thank you.
+              </span>
```
Comment (frederikebruendermann) [inline src/pages/CropDetailsPage.js:93]: very nice! It´s good to give this information to the user🚀

File: src/pages/CropDetailsPage.spec.js [hunk 2]
```diff
@@ -1,28 +1,32 @@
...
+              name: 'Strawberry',
+              botanicalName: 'botanical name',
+              sun: 'sun',
+              spread: 60,
+              rowSpace: 60,
+              details: 'details',
+            },
+          }}
+        />
+      </MemoryRouter>
```
Comment (frederikebruendermann) [inline src/pages/CropDetailsPage.spec.js:29]: 👍🏼

File: src/pages/ResultsPage.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+          </>
+        ) : (
+          <Paragraph>
+            No crop found.
+            <br />
+            Please try again and type at least 3 letters.
+          </Paragraph>
+        )}
+      </Output>
+    </div>
```
Comment (frederikebruendermann) [inline src/pages/ResultsPage.js:44]: Good work! UseParams is new for me, but it looks quite helpful to me, to make the code cleaner.

File: src/pages/SearchPage.js [hunk 4]
```diff
@@ -1,76 +1,74 @@
 import PropTypes from 'prop-types'
-import { useState } from 'react'
+import { useHistory } from 'react-router-dom'
 import styled from 'styled-components/macro'
 import Button from '../components/Button'
-import CropItem from '../components/CropItem'
 import Header from '../components/Header'
+import BackgroundDesktop from '../images/vegetable-frame-desktop.png'
+import BackgroundMobile from '../images/vegetable-frame-mobile.png'
```
Comment (frederikebruendermann) [inline src/pages/SearchPage.js:7]: 🚀

**Review by frederikebruendermann — APPROVED** [review]
Well done 🌼 It´s pretty nice to see new improvements on every of your PullRequests 🌟

**Pull request merged**